### PR TITLE
feat: add types to `RadioTile`

### LINKED ISSUE #13572: Add TypeScript types to RadioTile

### PR DESCRIPTION
**Pull request by cuppajoey**
Closes #13572 

This PR supersedes #14901 and addresses the type errors found in review. 

Converts `RadioTile` to Typescript. Add/exports newly added types.

#### Changelog

**New**

- Rename `RadioTile.js` → `RadioTile.tsx` and `RadioTile/index.js` to `RadioTile/index.ts`.
- Introduce and export interface `RadioTileProps`.

**Changed**

- Normalize PropType descriptions with similar components. The previous descriptions were inconsistent and did not match the format of other similar components (e.g., `TileGroup`, `RadioButton`, `Select`, etc..).

For example, most other PropType descriptions begin with "Provide an optional..." or "Specify whether 'X' should be disabled".

#### Testing / Reviewing

- [ ] Verify new types match the component's `PropTypes`.
- [ ] Check for Type errors.
- [ ] Confirm new `PropType` descriptions match the common format of similar components.

### CODE REVIEW

File: packages/react/src/components/RadioTile/RadioTile.tsx [hunk 1]
```diff
@@ -99,27 +164,27 @@ const RadioTile = React.forwardRef(function RadioTile(
 
 RadioTile.propTypes = {
```
Comment (cuppajoey) [inline packages/react/src/components/RadioTile/RadioTile.tsx:165]: I took the liberty of updated these prop descriptions to better match the writing style of other components (e.g., `TileGroup`, `Select`, etc..). If you feel this is out of scope for this PR, I don't mind reverting these changes.

File: packages/react/src/components/RadioTile/index.ts [hunk 2]
```diff
@@ -5,7 +5,9 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import RadioTile from './RadioTile';
+import RadioTile, { RadioTileProps } from './RadioTile';
 
 export default RadioTile;
 export { RadioTile };
+
+export type { RadioTileProps };
```
Comment (cuppajoey) [inline packages/react/src/components/RadioTile/index.ts:13]: I noticed that many of the recently added types don't export their interface. I'm not sure if there is a good reason not to, but I decided to include it.

This matches the [original implementation from DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/carbon-components-react/lib/components/RadioTile/RadioTile.d.ts) and the current [export for `Checkbox`](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/Checkbox/index.ts)

**Comment by cuppajoey** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/react/src/components/RadioTile/RadioTile.tsx [hunk 3]
```diff
@@ -21,12 +21,73 @@ import { noopFn } from '../../internal/noopFn';
...
+   */
+  disabled?: boolean;
+
+  /**
+   * Provide a unique id for the underlying `<input>`.
+   */
+  id?: string;
+
+  /**
+   * `true` to use the light version. For use on $ui-01 backgrounds only.
```
Comment (nchevsky) [inline packages/react/src/components/RadioTile/RadioTile.tsx]: I believe [`ui-01` has become `layer-01`](https://github.com/carbon-design-system/carbon/blob/main/docs/migration/v11.md#design-tokens) in Carbon 11.

```suggestion
   * `true` to use the light version. For use on $layer-01 backgrounds only.
```

**Review by tay1orjones — APPROVED** [review]
Looks good, thank you! 🙏 I think re-running the all contributors script should fix the conflicts.

**Comment by cuppajoey** [comment]
@tw15egan I'm going to push up one more commit to fix the description for `props.light`. Will that conflict with what you are doing?

**Comment by tw15egan** [comment]
@cuppajoey No thats fine, I just updated the `all-contributors` for you 👍

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅
